core/src/impl/Kokkos_Core.cpp

```diff
@@ -674,6 +674,15 @@ void post_initialize_internal(const Kokkos::InitializationSettings& settings) {
       settings.get_print_configuration()) {
     ::Kokkos::print_configuration(std::cout);
   }
+// In order to support setting an atexit hook for Kokkos::finalize
+// We need to ensure that the Cuda deep_copy instance is not destroyed
+// before that atexit hook is getting called.
+// Thus we create the static instance here, so that it will be deallocated
+// after the potential atexit call.
+// This is neccessary since we will access that instance in Kokkos::finalize
+#ifdef KOKKOS_ENABLE_CUDA
+  (void)::Kokkos::Impl::cuda_get_deep_copy_space(true);
```

Comment:
Why can't this be in `CUDA::impl_initialize()`?

Reply:
Ah right we have that thing. It can't be called inside the CudaInstance::initialize because that turns out to be recursive, but we can call it in Cuda::impl_initialize, I overlooked that call.